modules/CustomView/CustomView.php

```diff
@@ -724,12 +725,34 @@ public function getAdvFilterByCvid($cvid) {
 				}
 				$uitype = getUItypeByFieldName($this->customviewmodule, $col[2]);
 				if (($col[1]=='smownerid' || $col[1]=='smcreatorid' || $col[1]=='modifiedby' || $uitype==101)
-					&& $advfilterval=='current_user' && (empty($_REQUEST['action']) || $_REQUEST['action']!='CustomView') && empty($_REQUEST['record'])
+					&& ($advfilterval=='current_user' || $advfilterval=='current_userandgroup') && (empty($_REQUEST['action']) || $_REQUEST['action']!='CustomView') && empty($_REQUEST['record'])
 				) {
-					$advfilterval = trim($current_user->first_name.' '.$current_user->last_name);
+					if ($advfilterval=='current_user') {
+						$advfilterval = trim($current_user->first_name.' '.$current_user->last_name);
+					} else {
+						global $adb;
+						$result = $adb->pquery('select roleid from vtiger_users2group inner join vtiger_group2role on vtiger_users2group.groupid = vtiger_group2role.groupid where userid=?', array($current_user->id));
```

Comment:
should we use `select distinct roleid from ....` here to reduce duplicate values?